modules/Oculus_module/src/RobotControlHelper.cpp

```diff
@@ -0,0 +1,300 @@
+        return false;
+    }
+
+    m_desiredPositionInDegrees.resize(m_actuatedDOFs);
+    m_positionFeedbackInDegrees.resize(m_actuatedDOFs);
+    m_positionFeedbackInRadians.resize(m_actuatedDOFs);
+    m_minJointsPosition.resize(m_actuatedDOFs);
+    m_maxJointsPosition.resize(m_actuatedDOFs);
+
+    // check if the robot is alive
+    bool okPosition = false;
+    for (int i = 0; i < 10 && !okPosition; i++)
+    {
+        okPosition = m_encodersInterface->getEncoders(m_positionFeedbackInDegrees.data());
+
+        if (!okPosition)
+            yarp::os::Time::delay(0.1);
+    }
+    if (!okPosition)
+    {
+        yError() << "[configure] Unable to read encoders (position).";
+        m_isActive = false;
+        return false;
+    }
+
+    for (int i = 0; i < m_actuatedDOFs; i++)
+        // get position limits
+        if (!m_limitsInterface->getLimits(i, &m_minJointsPosition(i), &m_maxJointsPosition(i)))
+        {
+            yError() << "[configure] Unable get joints position limits.";
+            m_isActive = false;
+            return false;
+        }
+
+    return true;
+}
+
+bool RobotControlHelper::switchToControlMode(const int& controlMode)
+{
+    // if the part is not active skip
+    if (!m_isActive)
+        return true;
+
+    // check if the control interface is ready
+    if (!m_controlModeInterface)
+    {
+        yError() << "[RobotControlHelper::switchToControlMode] ControlMode I/F not ready.";
+        return false;
+    }
+
+    // set the control interface
+    std::vector<int> controlModes(m_actuatedDOFs, controlMode);
+    if (!m_controlModeInterface->setControlModes(controlModes.data()))
+    {
+        yError()
+            << "[RobotControlHelper::switchToControlMode] Error while setting the controlMode.";
+        return false;
+    }
+    return true;
+}
+
+bool RobotControlHelper::setDirectPositionReferences(const yarp::sig::Vector& desiredPosition)
+{
+    if (!m_isActive)
+        return true;
+
+    if (m_positionDirectInterface == nullptr)
+    {
+        yError()
+            << "[RobotControlHelper::setDirectPositionReferences] PositionDirect I/F not ready.";
+        return false;
+    }
+
+    if (desiredPosition.size() != m_actuatedDOFs)
+    {
+        yError() << "[RobotControlHelper::setDirectPositionReferences] Dimension mismatch between "
+                    "desired position vector and the number of controlled joints.";
+        return false;
+    }
+
+    double threshold = 5;
+    for (int i = 0; i < m_actuatedDOFs; i++)
+    {
+        m_desiredPositionInDegrees(i) = iDynTree::rad2deg(desiredPosition(i));
+
+        // check if the position is outside the limit
+        if (m_desiredPositionInDegrees(i) <= m_minJointsPosition(i))
+            m_desiredPositionInDegrees(i) = m_minJointsPosition(i) + threshold;
+        else if (m_desiredPositionInDegrees(i) >= m_maxJointsPosition(i))
+            m_desiredPositionInDegrees(i) = m_maxJointsPosition(i) - threshold;
+    }
```

Comment:
The references should be already clamped at firmware level. In this way you don't need to resort to the limits in this method (i.e. it is not a problem if some joints are in the "not configured" status).

Reply:
Done in 185a6dd